pkg/engine/context/loaders/imagedata.go

```diff
@@ -107,6 +108,10 @@ func (idl *imageDataLoader) fetchImageDataMap(client registryclient.Client, ref
 	if err != nil {
 		return nil, err
 	}
+	parsedRef, err := name.ParseReference(ref)
```

Comment:
Was this a cherry-pick from some upstream commit?

Reply:
Not an exact cherry pick, needed some changes to incorporate for rekor upgrade without upgrading cosign to 2.0. This came in for the same. Here this is for `go-containerregistry` upgrade.

Reply:
@vishal-chdhry - needs your expertise here :)

Reply:
Yes, this is a valid change, in older version of gcr, they had name.reference in the descriptors, but they removed it from gcr versions >0.15, now we have to parse the ref ourselves. We have the same change in OSS Kyverno.